packages/next/src/server/getProps.tsx

```diff
@@ -78,8 +78,9 @@ export async function getProps<
       async () => {
         const { cacheSnapshot: coreSnapshot } = await client.prepareReactRender(
           <RouterContext.Provider
-            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any,react/jsx-no-constructed-context-values
```

Comment:
There shouldn't be a need to use `useMemo` here as this call happens on the server and this hook is not available to use.